src/Stryker.Core/Stryker.Core/Mutants/MutantOrchestrator.cs

```diff
@@ -121,27 +113,140 @@ public SyntaxNode Mutate(SyntaxNode currentNode)
                     }
                     return ifStatement.ReplaceNode(ifStatement.Statement, Mutate(ifStatement.Statement));
                 }
-                return MutateWithIfStatements(statement);
+                else if (currentNode is ForStatementSyntax forStatement)
+                {
+                    // for needs special treatments for its incrementors
+                    StatementSyntax newNode = forStatement;
+                    foreach (var incrementor in forStatement.Incrementors)
+                    {
+                        var mutants = FlatMutate(incrementor);
+                        foreach (var mutant in mutants)
+                        {
+                            var newFor = ApplyMutant(forStatement, mutant);
+
+                            newNode = MutantPlacer.PlaceWithIfStatement(statement, newFor, mutant.Id);
+                        }
+                    }
+
+                    var originalFort = forStatement;
+                    if (newNode != originalFort)
+                    {
+                        originalFort = (ForStatementSyntax) ((BlockSyntax)((IfStatementSyntax) newNode).Else.Statement).Statements[0];
+                    }
+                    forStatement = MutateInPlace(originalFort, originalFort.Condition);
+                    forStatement = MutateInPlace(forStatement, forStatement.Statement);
+                    return newNode.ReplaceNode(originalFort, forStatement);
+                }
+                else
+                {
+                    result = MutateWithIfStatements(statement);
+                }
+
+                if (result != statement)
+                {
+                    var node = ((result as IfStatementSyntax).Else.Statement as BlockSyntax).Statements[0];
```

Comment:
Same here, tracknodes and getCurrentNode would be better here.

Reply:
Fixed.
Again, thanks for the pointer.
Only **for** statement remains to be done.